resource-managers/mesos/src/main/scala/org/apache/spark/deploy/rest/mesos/MesosRestServer.scala

```diff
@@ -139,20 +141,40 @@ private[mesos] class MesosSubmitRequestServlet(
       responseServlet: HttpServletResponse): SubmitRestProtocolResponse = {
     requestMessage match {
       case submitRequest: CreateSubmissionRequest =>
-        val driverDescription = buildDriverDescription(submitRequest)
-        val s = scheduler.submitDriver(driverDescription)
-        s.serverSparkVersion = sparkVersion
-        val unknownFields = findUnknownFields(requestMessageJson, requestMessage)
-        if (unknownFields.nonEmpty) {
-          // If there are fields that the server does not know about, warn the client
-          s.unknownFields = unknownFields
+        try {
+          val driverDescription = buildDriverDescription(submitRequest)
```

Comment:
While exceptions have been used in `buildDriverDescription` before, this approach doesn't look quite right. When we have a method returning value or throwing an error it's a natural use case for Scala's `Either` or `Try`. On the other hand, there're no usages of `Either` or `Try` in `mesos` package. On the other hand, having `try-catch` block here is a good defensive approach which can save us from surprises.

Given the status of the codebase for Mesos Scheduler in general, a more severe refactoring is needed so I'd say, let's keep the current code as is.